src/algorithms/PVT/libs/rinex_printer.cc

```diff
@@ -296,1263 +359,616 @@ Rinex_Printer::~Rinex_Printer()
-                            rinex_nav_header(navFile, pvt_solver->gps_iono, pvt_solver->gps_utc_model, gps_ephemeris_iter->second);
-                            output_navfilename.push_back(navfilename);
-                            log_rinex_nav(navFile, pvt_solver->gps_cnav_ephemeris_map);
-                            d_rinex_header_written = true;  // do not write header anymore
+                            // we have Galileo ephemeris, maybe from assistance
+                            rinex_obs_header(obsFile, gps_ephemeris_iter->second, galileo_ephemeris_iter->second, rx_time, signal);
+                            rinex_nav_header(navMixFile, pvt_solver->gps_iono, pvt_solver->gps_utc_model, gps_ephemeris_iter->second, pvt_solver->galileo_iono, pvt_solver->galileo_utc_model);
+                            output_navfilename.push_back(navMixfilename);
+                            log_rinex_nav(navMixFile, pvt_solver->gps_ephemeris_map, pvt_solver->galileo_ephemeris_map);
                         }
-                    break;
-                case 8:  // GPS L1 + GPS L5
-                    if ((gps_ephemeris_iter != pvt_solver->gps_ephemeris_map.cend()) && (gps_cnav_ephemeris_iter != pvt_solver->gps_cnav_ephemeris_map.cend()))
+                    else
                         {
-                            const std::string signal("1C L5");
-                            rinex_obs_header(obsFile, gps_ephemeris_iter->second, gps_cnav_ephemeris_iter->second, rx_time, signal);
+                            // we do not have galileo ephemeris, print only GPS data
+                            rinex_obs_header(obsFile, gps_ephemeris_iter->second, rx_time);
                             rinex_nav_header(navFile, pvt_solver->gps_iono, pvt_solver->gps_utc_model, gps_ephemeris_iter->second);
                             output_navfilename.push_back(navfilename);
                             log_rinex_nav(navFile, pvt_solver->gps_ephemeris_map);
-                            d_rinex_header_written = true;  // do not write header anymore
                         }
-                    break;
-                case 9:  // GPS L1 C/A + Galileo E1B
-                    if ((galileo_ephemeris_iter != pvt_solver->galileo_ephemeris_map.cend()) && (gps_ephemeris_iter != pvt_solver->gps_ephemeris_map.cend()))
+                }
+            else if (has_beidou && beidou_dnav_ephemeris_iter != pvt_solver->beidou_dnav_ephemeris_map.cend())
+                {
+                    rinex_obs_header(obsFile, beidou_dnav_ephemeris_iter->second, rx_time, signal);
+                    // Not implemented for beidou
+                }
+            else
+                {
+                    rinex_header_written = false;
+                }
+
+            d_rinex_header_written = rinex_header_written;
+        }
+
+    if (d_rinex_header_written && flag_write_RINEX_obs_output)  // The header is already written, we can now log the navigation message data
```

Comment:
From here sometimes we update the headers without having any ephemeris and sometime we don't, not sure what the logic here